noah/src/main/scala/edu/uci/ics/cloudberry/noah/feed/TwitterUserStreamDriver.java

```diff
@@ -0,0 +1,78 @@
+package edu.uci.ics.cloudberry.noah.feed;
+
+import com.twitter.hbc.ClientBuilder;
+import com.twitter.hbc.core.Client;
+import com.twitter.hbc.core.Constants;
+import com.twitter.hbc.core.endpoint.StatusesFilterEndpoint;
+import com.twitter.hbc.core.processor.StringDelimitedProcessor;
+import com.twitter.hbc.httpclient.auth.Authentication;
+import com.twitter.hbc.httpclient.auth.OAuth1;
+import org.kohsuke.args4j.CmdLineException;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+public class TwitterUserStreamDriver {
+
+    public void run(Config config) throws IOException, CmdLineException {
+
+        List<Long> usersID = Arrays.asList(CmdLineAux.parseID(config.getTrackUsers()));
+
+        Client twitterClient;
+        boolean isConnected = false;
+        BlockingQueue<String> queue = new LinkedBlockingQueue<String>(10000);
+        StatusesFilterEndpoint endpoint = new StatusesFilterEndpoint();
+        if (!usersID.isEmpty()) {
+            endpoint.followings(usersID);
+        }
+        Authentication auth = new OAuth1(config.getConsumerKey(), config.getConsumerSecret(), config.getToken(),
+                config.getTokenSecret());
+
+        // Create a new BasicClient. By default gzip is enabled.
+        twitterClient = new ClientBuilder()
+                .hosts(Constants.STREAM_HOST)
+                .endpoint(endpoint)
+                .authentication(auth)
+                .processor(new StringDelimitedProcessor(queue))
+                .build();
+
+        // Establish a connection
+        BufferedWriter bw = CmdLineAux.createWriter("Tweet_UserStream_");
```

Comment:
Probably stick to the `gz` format as well? Then the downstream process could be unified.